### PR TITLE
trino: Support CertificateAuthentication in the trino hook

### PR DESCRIPTION
**Pull request by c2zwdjnlcg**
Allows authenticating to trino clusters with mtls

### CODE REVIEW

File: airflow/providers/trino/hooks/trino.py [hunk 1]
```diff
@@ -99,12 +99,19 @@ def get_conn(self) -> Connection:
         extra = db.extra_dejson
         auth = None
         user = db.login
-        if db.password and extra.get('auth') == 'kerberos':
-            raise AirflowException("Kerberos authorization doesn't support password.")
+        if db.password and extra.get('auth') in ('kerberos', 'certs'):
+            raise AirflowException(
+                "The '{}' authorization type doesn't support password.".format(extra.get('auth'))
+            )
```
Comment (uranusjr) [inline airflow/providers/trino/hooks/trino.py]:
```suggestion
            raise AirflowException(
                f"The {extra.get('auth')!r} authorization type doesn't support password."
            )
```

Comment (c2zwdjnlcg) [reply airflow/providers/trino/hooks/trino.py]: Thanks @uranusjr, updated. Can you approve the tests workflow run

**Comment by c2zwdjnlcg** [comment]
@mik-laj  would you be able to take a look and approve the workflow runs? When marking the last review as updated I accidentally removed you as a reviewer

**Review by uranusjr — APPROVED** [review]

**Comment by c2zwdjnlcg** [comment]
Thanks for the review @uranusjr . Is this mergable or do we need @mik-laj to have a look as well?

**Review by potiuk — APPROVED** [review]

**Pull request merged**